src/main/java/org/broadinstitute/hellbender/utils/read/ReadCoordinateComparator.java

```diff
@@ -54,10 +60,9 @@ private int compareCoordinates( final GATKRead first, final GATKRead second ) {
         final int firstRefIndex = ReadUtils.getReferenceIndex(first, header);
         final int secondRefIndex = ReadUtils.getReferenceIndex(second, header);
 
-        if ( first.isUnmapped() ) {
-            return second.isUnmapped() ? 0 : 1;
-        }
-        else if ( second.isUnmapped() ) {
+        if ( firstRefIndex == SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX ) {
+            return secondRefIndex == SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX ? 0 : 1;
+        } else if ( secondRefIndex == SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX ) {
```

Comment:
This change seems misguided to me. We are getting the reference indices from `ReadUtils.getReferenceIndex()` in this comparator, which calls `GATKRead.isUnmapped()`:

    public static int getReferenceIndex( final GATKRead read, final SAMFileHeader header ) {
        if ( read.isUnmapped() ) {
            return SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX;
        }

        return header.getSequenceIndex(read.getContig());
    }

So your change here actually accomplishes nothing, and just makes the code uglier.